overlord/restart/restart_context.go

```diff
@@ -0,0 +1,82 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package restart
+
+import (
+	"github.com/snapcore/snapd/boot"
+	"github.com/snapcore/snapd/bootloader"
+	"github.com/snapcore/snapd/snap"
+)
+
+type RestartContext struct {
+	SnapName        string      `json:"snap-name,omitempty"`
+	RestartType     RestartType `json:"restart-type,omitempty"`
+	RebootArguments string      `json:"reboot-args,omitempty"`
+}
+
+// RestartContext implements the boot.RebootBootloader interface to support
+// the usage of reboot arguments in reboot.go (Reboot). This must be done in order
+// to correctly fake the boot.RebootInfo passed down to the reboot code. The only
+// function used from this interface is GetRebootArguments().
+func (rt *RestartContext) GetBootVars(names ...string) (map[string]string, error) {
+	panic("restart.RestartContext: unexpected call to GetBootVars")
+}
+
+func (rt *RestartContext) SetBootVars(values map[string]string) error {
+	panic("restart.RestartContext: unexpected call to SetBootVars")
+}
+
+func (rt *RestartContext) Name() string {
+	panic("restart.RestartContext: unexpected call to Name")
+}
+
+func (rt *RestartContext) Present() (bool, error) {
+	panic("restart.RestartContext: unexpected call to Present")
+}
+
+func (rt *RestartContext) InstallBootConfig(gadgetDir string, opts *bootloader.Options) error {
+	panic("restart.RestartContext: unexpected call to InstallBootConfig")
+}
+
+func (rt *RestartContext) ExtractKernelAssets(s snap.PlaceInfo, snapf snap.Container) error {
+	panic("restart.RestartContext: unexpected call to ExtractKernelAssets")
+}
+
+func (rt *RestartContext) RemoveKernelAssets(s snap.PlaceInfo) error {
+	panic("restart.RestartContext: unexpected call to RemoveKernelAssets")
+}
+
+// GetRebootArguments returns the previously stored reboot arguments.
+func (rt *RestartContext) GetRebootArguments() (string, error) {
+	return rt.RebootArguments, nil
+}
+
+func (rt *RestartContext) setParameters(snapName string, restartType RestartType, rebootInfo *boot.RebootInfo) error {
```

Comment:
if we call the structure RestartParameters this should then be called just init as we use for some unexported structures in devicestate

Reply:
yes, I felt a bit redundancy in having it named `setParameters` - renamed to `init`